todos.go

```diff
@@ -0,0 +1,109 @@
+package gitlab
+
+import "time"
+import "fmt"
+
+// TodosService handles communication with the todos related methods of
+// the Gitlab API.
+//
+// GitLab API docs:
+// https://gitlab.com/gitlab-org/gitlab-ce/blob/8-16-stable/doc/api/todos.md
+type TodosService struct {
+	client *Client
+}
+
+// Todo represents a GitLab todo.
+//
+// GitLab API docs:
+// https://gitlab.com/gitlab-org/gitlab-ce/blob/8-16-stable/doc/api/todos.md
+type Todo struct {
+	ID      int `json:"id"`
+	Project struct {
+		ID                int    `json:"id"`
+		HTTPURLToRepo     string `json:"http_url_to_repo"`
+		WebURL            string `json:"web_url"`
+		Name              string `json:"name"`
+		NameWithNamespace string `json:"name_with_namespace"`
+		Path              string `json:"path"`
+		PathWithNamespace string `json:"path_with_namespace"`
+	} `json:"project"`
+	Author struct {
+		ID        int    `json:"id"`
+		Name      string `json:"name"`
+		Username  string `json:"username"`
+		State     string `json:"state"`
+		AvatarURL string `json:"avatar_url"`
+		WebURL    string `json:"web_url"`
+	} `json:"author"`
+	ActionName string      `json:"action_name"`
+	TargetType string      `json:"target_type"`
+	Target     interface{} `json:"target"`
+	TargetURL  string      `json:"target_url"`
+	Body       string      `json:"body"`
+	State      string      `json:"state"`
+	CreatedAt  *time.Time  `json:"created_at"`
+}
+
+func (t Todo) String() string {
+	return Stringify(t)
+}
+
+// ListTodosOptions represents the available ListTodos() options.
+//
+// GitLab API docs:
+// https://docs.gitlab.com/ce/api/todos.html#get-a-list-of-todos
+type ListTodosOptions struct {
+	Action    string `url:"action,omitempty" json:"action,omitempty"`
+	AuthorID  int    `url:"author_id,omitempty" json:"author_id,omitempty"`
+	ProjectID int    `url:"project_id,omitempty" json:"project_id,omitempty"`
+	State     string `url:"state,omitempty" json:"state,omitempty"`
+	Type      string `url:"type,omitempty" json:"type,omitempty"`
+}
```

Comment:
These should all be pointer values. See all other option structs in this package, they all have pointer fields. This is needed to be able to make a distinction between a types zero value and no value so that the `omitempty` works as expected.